clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2064,9 +2064,18 @@ bool Sema::CheckAttrNoArgs(const ParsedAttr &Attrs) {
   return false;
 }
 
+static bool isAttrValidForSYCLDevice(ASTContext &Context, const ParsedAttr &AL)
+{
+   return Context.getLangOpts().SYCLIsDevice &&
+          Context.getTargetInfo().getTriple().isOSWindows() &&
+          (AL.getKind() == ParsedAttr::AT_DLLExport ||
+           AL.getKind() == ParsedAttr::AT_DLLImport);
+}
```

Comment:
This function returns true ONLY if (SYCL && Windows && (dllexport || dllimport))
My concern here is the name of the function, which is pretty general, while the func checks only DLL attrs. 
Wouldn't it be better than to name it something like isValidDllAttrForSYCL()?

Reply:
That is good idea.  That was my intended, like you said this may extended for more attribute checking in Linux as well as window.  But I am not sure, since I am not the one working for sycl.  
Thanks.

Reply:
Ok. Thank you. I mark this as resolved.